atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleAlterTableTest.java

```diff
@@ -0,0 +1,180 @@
+    private static final String COLUMN_NAME = "variable";
+
+    private static final TableMetadata EXPECTED_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.STRING)
+            .build();
+
+    private static final TableMetadata OLD_TABLE_METADATA = TableMetadata.builder()
+            .singleRowComponent("name", ValueType.STRING)
+            .singleNamedColumn("var", COLUMN_NAME, ValueType.FIXED_LONG)
+            .build();
+
+    private static final Cell DEFAULT_CELL = createCell("foo", "bar");
+    private static final Cell DEFAULT_CELL_2 = createCell("something", "bar");
+
+    private static final byte[] DEFAULT_VALUE = PtBytes.toBytes("amazing");
+    private static final long TIMESTAMP = 1L;
+
+    private ConnectionManagerAwareDbKvs defaultKvs;
+    private ConnectionSupplier connectionSupplier;
+    private OracleTableNameGetter oracleTableNameGetter;
+
+    @Before
+    public void setup() {
+        defaultKvs = (ConnectionManagerAwareDbKvs) TRM.getDefaultKvs();
+        connectionSupplier = DbKvsOracleTestSuite.getConnectionSupplier(defaultKvs);
+        oracleTableNameGetter = OracleTableNameGetterImpl.createDefault((OracleDdlConfig) CONFIG.ddl());
+    }
+
+    @After
+    public void tearDown() {
+        defaultKvs.dropTables(defaultKvs.getAllTableNames());
+        connectionSupplier.close();
+    }
+
+    @Test
+    public void whenConfiguredAlterTableToMatchMetadataAndOldDataIsStillReadable() {
+        defaultKvs.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        writeData(defaultKvs);
+        assertThatDataCanBeRead(defaultKvs);
+        dropOverflowColumn(defaultKvs);
+        defaultKvs.putMetadataForTable(TABLE_REFERENCE, OLD_TABLE_METADATA.persistToBytes());
+        assertThatThrownBy(() -> fetchData(defaultKvs));
+        ConnectionManagerAwareDbKvs workingKvs = ConnectionManagerAwareDbKvs.create(CONFIG_WITH_ALTER);
+        workingKvs.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        assertThatDataCanBeRead(workingKvs);
+        assertThatOverflowColumnExists(workingKvs);
+        assertThatDataCanBeWritten(workingKvs);
+    }
+
+    @Test
+    public void whenConfiguredAlterTableDoesNothingWhenMatching() {
+        ConnectionManagerAwareDbKvs kvsWithAlter = ConnectionManagerAwareDbKvs.create(CONFIG_WITH_ALTER);
+        kvsWithAlter.createTable(TABLE_REFERENCE, EXPECTED_TABLE_METADATA.persistToBytes());
+        writeData(kvsWithAlter);
+        assertThatDataCanBeRead(defaultKvs);
+    }
+
+    private void assertThatDataCanBeWritten(ConnectionManagerAwareDbKvs kvs) {
+        assertThatCode(() -> kvs.put(TABLE_REFERENCE, Map.of(DEFAULT_CELL_2, DEFAULT_VALUE), TIMESTAMP))
+                .doesNotThrowAnyException();
+    }
+
+    private void assertThatOverflowColumnExists(ConnectionManagerAwareDbKvs kvs) {
+        assertThatCode(() -> {
```

Comment:
We don't need the doesNotThrowAnyException wrapper around this - it's especially off since you're asserting inside of it!

Reply:
It's just b/c `OracleTableNameGetter::getInternalShortTableName` throws `TableMappingNotFoundException` explicitly, so I figured instead of rethrowing/adding to the statement, we'd just check that it doesn't assert

Reply:
happy to go either way!

Reply:
Given you later do a try catch and rethrow, I'd just recommend either doing the same instead of the assertCode wrapper, or adding a utility method like (in dodgy java TM)

```runRethrowingTableMappingException(Runnable) {
 try {
runnable.run()
} catch (TME) {
 throw new RuntimeException(e);
}```